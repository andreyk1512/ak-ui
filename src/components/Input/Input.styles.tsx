import { css } from '@emotion/react';
import * as styles from '../../styles';

export const inputWrapperStyles = () => css`
  display: inline-flex;
  align-items: center;
  background-color: ${styles.color.aliceBlue};
  border-radius: 6px;
  color: ${styles.color.uclaBlue};
  width: 100%;
`;

export const inputStyles = () => css`
  display: block;
  height: 45px;
  border: none;
  box-sizing: border-box;
  padding: 0 15px;
  font-size: ${styles.fontSize.regular};
  background-color: transparent;
  overflow: hidden;
  color: inherit;
  width: 100%;

  &:focus {
    color: ${styles.color.yankeesBlue};
    outline: none;
  }
`;

export const suffixIconStyles = () => css`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  margin-inline-end: 15px;
  color: inherit;
  max-width: 24px;
  max-height: 24px;
`;
