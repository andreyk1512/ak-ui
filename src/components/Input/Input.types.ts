import { InputHTMLAttributes, ReactNode } from 'react';

export type InputProps = InputHTMLAttributes<HTMLInputElement> & {
  suffixIcon?: ReactNode;
};
