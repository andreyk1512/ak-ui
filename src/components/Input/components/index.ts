export * from './Password';
export * from './TextArea';
export * from './File';
