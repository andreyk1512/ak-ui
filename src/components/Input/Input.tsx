import { Password, TextArea } from './components';
import { inputStyles, inputWrapperStyles, suffixIconStyles } from './Input.styles';
import { InputProps } from './Input.types';

export const Input = ({ suffixIcon, ...inputProps }: InputProps) => (
  <div css={inputWrapperStyles()}>
    <input {...inputProps} css={inputStyles()} />
    {suffixIcon ? <span css={suffixIconStyles()}>{suffixIcon}</span> : null}
  </div>
);

Input.Password = Password;
Input.TextArea = TextArea;
