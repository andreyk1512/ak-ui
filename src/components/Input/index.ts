export * from './Input';
export * from './Input.types';

import { Password, TextArea, File } from './components';
import { Input as OriginInput } from './Input';

export type InputProps = typeof OriginInput & {
  Password: typeof Password;
  TextArea: typeof TextArea;
  File: typeof File;
};

export const Input = OriginInput as InputProps;
Input.Password = Password;
Input.TextArea = TextArea;
Input.File = File;
