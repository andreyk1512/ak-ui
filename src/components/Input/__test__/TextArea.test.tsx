import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Input } from '../Input';

describe('TextArea', () => {
  it('should render the textarea element', () => {
    const { container } = render(<Input.TextArea />);

    const input = screen.getByRole('textbox');
    expect(input).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
