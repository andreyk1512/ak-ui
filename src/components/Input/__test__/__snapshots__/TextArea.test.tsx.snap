// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea should render the textarea element 1`] = `
<textarea
  class="css-1bkhsa2"
/>
`;
